lib/rubygems/test_case.rb

```diff
@@ -1319,9 +1319,18 @@ def self.rubybin
     end
   end
 
+  def self.escape_path(path)
+    if %r'\A[-+:/=@,.\w]+\z' =~ path
+      path
+    else
+      "\"#{path.gsub(/[`$"]/, '\\&')}\""
+    end
+  end
+
   @@ruby = rubybin
-  @@good_rake = "#{rubybin} \"#{File.expand_path('../../../test/rubygems/good_rake.rb', __FILE__)}\""
-  @@bad_rake = "#{rubybin} \"#{File.expand_path('../../../test/rubygems/bad_rake.rb', __FILE__)}\""
+  gempath = File.expand_path('../../../test/rubygems', __FILE__)
+  @@good_rake = "#{rubybin} #{escape_path(gempath+'/good_rake.rb')}"
+  @@bad_rake = "#{rubybin} #{escape_path(gempath+'/bad_rake.rb')}"
```

Comment:
Could `gempath+'/good_rake.rb'` be replaced with `File.join(gempath, 'good_rake.rb')` (and similar for bad_rake.rb)?